File: TestLBXAlertAction/ViewController.m
//
//  ViewController.m
//  TestLBXAlertAction
//
//  Created by lbxia on 15/11/12.
//  Copyright © 2015年 lbxia. All rights reserved.
//

#import "ViewController.h"
#import "LBXAlertAction.h"
#import "ModelViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (void)dosomething
{
    
}
- (IBAction)btnAlert:(id)sender {
    
    __weak __typeof(self) weakSelf = self;
    [LBXAlertAction showAlertWithTitle:@"标题"
                                   msg:@"提示消息内容"
                      buttonsStatement:@[@"取消",@"确认1",@"确认2",@"确认3",@"确认4",@"确认5",@"确认6"]
                           chooseBlock:^(NSInteger buttonIdx) {
                               
                               NSArray* array = @[@"取消",@"确认1",@"确认2",@"确认3",@"确认4",@"确认5",@"确认6"];
                               
                               //按钮按顺序输出
                               NSLog(@"u choose:%@",array[buttonIdx]);
                               
                               [weakSelf dosomething];
                           }];

}
- (IBAction)btnSheet:(id)sender {
    
    __weak __typeof(self) weakSelf = self;
    [LBXAlertAction showActionSheetWithTitle:@"标题"
                                     message:@"ios8系统之后才会显示本消息内容"
                           cancelButtonTitle:@"取消"
                      destructiveButtonTitle:@"destructive"
                            otherButtonTitle:@[@"items1",@"items2",@"items3"]
                                 chooseBlock:^(NSInteger buttonIdx) {
                                     
                                     //取消为0，destructiveButtonTitle从1开始，如果输入为nil，则otherButtonTitle从1开始，否则从2开始
                                     NSLog(@"%ld",buttonIdx);
                                     [weakSelf dosomething];
                                 }];    
}

- (IBAction)modelVC:(id)sender {
    
    ModelViewController *vc = [[ModelViewController alloc]init];
    
    [self presentViewController:vc animated:YES completion:nil];
}

- (BOOL)shouldAutorotate
{
    return YES;
}

- (UIInterfaceOrientationMask)supportedInterfaceOrientations
{
    return  UIInterfaceOrientationMaskPortrait;
}


@end

File: TestSwiftAlertAction/ViewController.swift
//
//  ViewController.swift
//  TestSwiftAlertAction
//
//  Created by lbxia on 16/6/17.
//  Copyright © 2016年 lbx. All rights reserved.
//

import UIKit


class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
      
    }

    @IBAction func testAlertView(_ sender: AnyObject)
    {

//        let alert = UIAlertView(title: "title", message: "message", delegate: nil, cancelButtonTitle: "cance", otherButtonTitles: "sure", "sure2")
//        
//        alert .show { (buttonIndex) in
//            
//            print(buttonIndex)
//        }
        
        let items = ["cancel","ok1","ok2"];
        
        AlertAction.showAlert(title: "title", message: "message", btnStatements:items )
        {
            (buttonIndex) in
            
            
            let items = ["cancel","ok1","ok2"];
            
            print(buttonIndex)
            
            print(items[buttonIndex])
          
        }
    }
    
    @IBAction func testSheetView(_ sender: AnyObject)
    {
        let destrucitve:String? = "destructive"
        
//        let destrucitve:String? = nil
        
        AlertAction.showSheet(title: "title", message: "ios8之后才会显示本条信息",  destructiveButtonTitle: destrucitve,cancelButtonTitle: "cancel", otherButtonTitles: ["other1","other2"]) { (buttonIdx, itemTitle) in
            
            /*
             经测试
             buttonIdx：  destructiveButtonTitle 为0， cancelButtonTitle 为1，otherButtonTitles按顺序增加
             如果destructiveButtonTitle 传入值为nil,那么 cancelButtonTitle 为0，otherButtonTitles按顺序增加
             
             或者按照itemTitle来判断用户点击那个按钮更稳妥
             */
            
            print(buttonIdx)
            print(itemTitle)
        }

    }
    
    func alertResult(_ buttonIndex:Int) -> Void
    {
        
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

